azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/collection/ResourcesInGroup.java

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.collection;
+
+public interface ResourcesInGroup<T, DefinitionT> extends
+    SupportsListing<T>,
+    SupportsCreating<DefinitionT>,
+    SupportsDeleting {}
```

Comment:
Given an Id we could extract the resourceGroup name and resource name and invoke getByGroup(groupName, name), but issue here is the Id provided by the user can be pointing to any resource not just a resource under the collection in conext, for example virtualMachines.getById(network.Id()).